synapseclient/table.py

```diff
@@ -397,16 +397,16 @@ def _convert_df_date_cols_to_datetime(
 
 
 def _csv_to_pandas_df(
-    filepath,
-    separator=DEFAULT_SEPARATOR,
-    quote_char=DEFAULT_QUOTE_CHARACTER,
-    escape_char=DEFAULT_ESCAPSE_CHAR,
-    contain_headers=True,
-    lines_to_skip=0,
-    date_columns=None,
-    list_columns=None,
-    rowIdAndVersionInIndex=True,
-    dtype=None,
+    filepath: str,
+    separator: str = DEFAULT_SEPARATOR,
+    quote_char: str = DEFAULT_QUOTE_CHARACTER,
+    escape_char: str = DEFAULT_ESCAPSE_CHAR,
+    contain_headers: bool = True,
+    lines_to_skip: int = 0,
+    date_columns: Optional[List[str]] = None,
+    list_columns: Optional[List[str]] = None,
+    rowIdAndVersionInIndex: bool = True,
+    dtype: Optional[Dict[str, Any]] = None,
     **kwargs,
 ):
```

Comment:
nit: Add a pandas df return type here

Reply:
I can't add this unfortunately because pandas is imported within the function definition so doing that causes an error

Reply:
OK this isn't a huge deal so I'll resolve this. I'm wondering why we're importing `pandas` this way instead of at the top, and why there's this extra test for importing pandas, it seems kind of odd. Tagging @BryanFauble in here for when he gets back

Reply:
I can chime in here, it's because pandas historically is an optional dependency of the client.

Reply:
Good to know!